lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -3706,9 +3706,11 @@ CheckedCastKind TypeChecker::typeCheckCheckedCast(Type fromType,
   else
     fromRequiresClass = fromType->mayHaveSuperclass();
   
-  // Casts between protocol metatypes only succeed if the type is existential.
+  // Casts between protocol metatypes only succeed if the type is existential
```

Comment:
Is the original comment here correct?  I read the code below as saying that metatype casts always *fail* if the type is existential.

Reply:
Humm, it really seems like that, so I think the comment is at the very least misleading... maybe It should say that `Casts between protocol metatypes only succeed if none of the types are existential` which makes sense
cc @xedin 

Thanks for pointing that out @tbkka :)